pkg/operator/ceph/operator.go

```diff
@@ -205,3 +208,25 @@ func (o *Operator) startSystemDaemons(clusterSpec *cephv1.ClusterSpec) error {
 	o.delayedDaemonsStarted = true
 	return nil
 }
+
+func (o *Operator) stopSystemDaemons() error {
+	if !o.delayedDaemonsStarted || o.clusterController.GetClusterCount() != 0 {
+		return nil
+	}
+	if o.operatorNamespace == "" {
+		return fmt.Errorf("Rook operator namespace is not provided. Expose it via downward API in the rook operator manifest file using environment variable %s", k8sutil.PodNamespaceEnvVar)
+	}
+	// TODO: Add removal of FlexVolume daemons
+	if !csi.CSIEnabled() {
+		logger.Infof("CSI driver is not enabled")
+		return nil
+	}
+
+	if err := csi.StopCSIDrivers(o.operatorNamespace, o.context.Clientset); err != nil {
```

Comment:
Why not call `DeleteCsiConfigMap()` directly? Looks like that's the only thing `StopCSIDrivers()` does.

Reply:
I used the callback more or less only to twiddle the o.delayedDaemonsStarted bit. While a bit silly, this is necessary so that the operator knows to start the daemons again if and when another cluster is created. I thought about exporting delatedDaemonsStarted, or moving it to the ClusterController, but preferred this to either of those solutions (though it is more code).

Reply:
I only did this to follow pattern with StartCSIDrivers, really. It lets me pass the operator namespace through, although we can absolutely just use os.Getenv(k8sutil.PodNamespaceEnvVar). I'm completely flexible on this point.

Reply:
I'm ok either way, it was just a nit. Feel free to change it or not. If it follows the existing pattern, probably best not to change it.